server/controller/rideRequest.js

```diff
@@ -19,64 +16,62 @@ class RideRequestController{
                     })
                 }else{
                     return res.status(200).json({
-                        message: 'All ride request have be fetch.',
-                        rideRequest
+                        success: true,
+                        requests: rideRequest
                     })
                 }
             }else{
                 return next(err)
             }
         })
     }
-    static postRideRequest (req, res, next) {
-        const text = "INSERT INTO rideRequests (accept, reject, userId, rideId, created_at) VALUES ($1, $2, $3, $4, Now()) RETURNING *";
+    static postRideRequest (req, res) {
+        const text = "INSERT INTO rideRequests (userId, rideId, created_at) VALUES ($1, $2, Now()) RETURNING *";
         const values = [
-            req.body.accept,
-            req.body.reject,
             req.decoded.userId,
             req.params.rideId
         ]
         client.query(text, values, (err, data) =>{
             if(err){
-                return res.status(400).json({
+                const duplicateKeyError = "riderequests_userid_rideid_key";
+                if(err.message.search(duplicateKeyError) !== -1){
+                    return res.status(500).json({
+                        success: false,
+                        message: 'Error: You can only request for a ride once.'
+                    })
+                }
+                return res.status(404).json({
```

Comment:
please i don't understand what you mean here, can you please explain further?

Reply:
Okay, what I mean here is that you just checked if there is an err, return 404 user not found. it is wrong because we would always get user not found whenever there is an error regardless of what the error might be.

Reply:
Okay, what I mean here is that you just checked if there is an err, return 404 user not found. it is wrong because we would always get user not found whenever there is an error regardless of what the error might be.